dts/bindings/misc/raspberrypi,pico-pio.yaml

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2023, TOKITA Hiroshi
+# SPDX-License-Identifier: Apache-2.0
+
+description: Raspberry Pi Pico PIO
+
+compatible: "raspberrypi,pico-pio"
+
+include: [base.yaml, pinctrl-device.yaml, reset-device.yaml]
+
+properties:
+  state-machine:
+    type: array
+    description: |
+      Specifies the id of state machines to use.
+      Persistently use the state machine with the specified Id.
+
+  state-machine-num:
+    type: int
+    description: |
+      Specifies the number of state machines to use.
+      Dynamically allocate this number of state machines from the not used state machine.
+      If the `state-machine property` is also specified,
+      preferentially use the IDs specified in the state-machine property.
+      If the value of this property is greater than the number of elements
+      in the state-machine property, dynamically allocate the missings.
+
+  pinmuxs:
+    type: phandle-array
+    description: |
+      A phandle array of pinmuxes used by the higher level driver. The role of
+      each pinmux is determined by `pinmux-names`. For example, if PIO1 is used
+      as a UART peripheral, `pinmuxs` will be set to `<&pio1_uart_rx>, <&pio1_uart_tx>`.
+
+  pinmux-names:
+    type: string-array
+    description: |
+      The names of the pinmuxes as expected by the higher level driver. For example,
+      if the PIO is used as a UART peripheral, `pinmux-names` will be set to `"rx", "tx"`.
```

Comment:
-1 on custom pinmux schemes

Reply:
Would the suggestion by @soburi [here](https://github.com/zephyrproject-rtos/zephyr/pull/44316#discussion_r1127166096) solve this?

Reply:
@gmarull 
Is this pointing out about that the name `"#pinmuxs-cells"` may use commonly, should not use here?
Does the concern solved if the name change to `pio-pinmuxs`?